pkg/access/rewrap.go

```diff
@@ -148,8 +167,11 @@ func (p *Provider) Handler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if requestBody.Algorithm == "ec:secp256r1" {
-		http.Error(w, "Unsupported Algorithm", http.StatusBadRequest)
-		slog.WarnContext(ctx, "Nano not implemented yet")
+		responseBytes, err := nanoTDFRewrap(requestBody)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
For now this might be sufficient:

```suggestion
			http.Error(w, "invalid request", http.StatusBadRequest)
  		slog.WarnContext(ctx, "nanotdf failure", "err", err)
  		return
```

But we may want to differentiate between error types which would be more like:

```
		if errors.As(err, &ErrCausedByInput) {
			http.Error(w, "invalid request", http.StatusBadRequest)
		} else if errors.As(err, &ErrAuthorization) {
			http.Error(w, "invalid auth", http.StatusNotAuthorized)
		} else {
			http.Error(w, "unknown error", http.StatusInternalServerError)
		}
```

where we define the const `ErrX`es above and just errs.Join to generate them with additional flavor text.